sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -181,3 +181,36 @@ def test_ipy_notebook():
 
         f.flush()
         assert_equal(json.load(f), example_nb.work_notebook)
+
+
+def test_save_figures():
+    """Test file naming when saving figures. Requires mayavi."""
+    import matplotlib.pylab as plt
```

Comment:
For the matplotlib backend to be set to Agg correctly you probably need the matplotlib.pyplot import to be after the gen_rst one though.